apps/meteor/client/views/omnichannel/directory/contacts/RemoveContactModal.tsx

```diff
@@ -0,0 +1,76 @@
+import { Box, Input } from '@rocket.chat/fuselage';
+import { useEffectEvent } from '@rocket.chat/fuselage-hooks';
+import { GenericModal } from '@rocket.chat/ui-client';
+import { useToastMessageDispatch, useEndpoint } from '@rocket.chat/ui-contexts';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import type { ReactElement, ChangeEvent } from 'react';
+import { useState, useId } from 'react';
+import { useTranslation } from 'react-i18next';
+
+type RemoveContactModalProps = {
+	_id: string;
+	name: string;
+	channelsCount: number;
+	onClose: () => void;
+};
+
+const RemoveContactModal = ({ _id, name, channelsCount, onClose }: RemoveContactModalProps): ReactElement => {
+	const { t } = useTranslation();
+	const [text, setText] = useState<string>('');
+
+	const queryClient = useQueryClient();
+	const removeContact = useEndpoint('POST', '/v1/omnichannel/contacts.delete');
+	const dispatchToast = useToastMessageDispatch();
+	const contactDeleteModalId = useId();
+
+	const handleSubmit = useEffectEvent((event: ChangeEvent<HTMLFormElement>): void => {
+		event.preventDefault();
+		removeContactMutation.mutate();
+	});
+
+	const removeContactMutation = useMutation({
+		mutationFn: () => removeContact({ contactId: _id }),
+		onSuccess: async () => {
+			dispatchToast({ type: 'success', message: t('Contact_has_been_deleted') });
+			await Promise.all([
+				queryClient.invalidateQueries({
+					queryKey: ['current-contacts'],
+				}),
+				queryClient.invalidateQueries({
+					queryKey: ['getContactsByIds', _id],
+				}),
+			]);
+			onClose();
+		},
+		onError: (error) => {
+			dispatchToast({ type: 'error', message: t(error.message, { defaultValue: t('error-contact-something-went-wrong') }) });
+		},
+	});
+
+	return (
+		<GenericModal
+			wrapperFunction={(props) => <Box is='form' onSubmit={handleSubmit} {...props} />}
+			onCancel={onClose}
+			confirmText={t('Delete')}
+			title={t('Delete_Contact')}
+			onClose={onClose}
+			variant='danger'
+			confirmDisabled={text !== t('Delete').toLowerCase()}
```

Comment:
I am concerned with a _(unlikely but possible)_ mismatch in cases where `Are_you_sure_delete_contact` i18n expression is translated in a language (e.g pt-BR) but `Delete` i18n exp is not. It could cause the description to say something like "escreva 'Deletar' abaixo" while `Delete` is missing its translation and fallbacks to english, causing this check to always fail.

A possible solution could be to add the confirmation word as a parameter of the description, that way they are aways the same.

Reply:
Stumbled on this comment, the solution might be

```
"Are_you_sure_delete_contact": "Are you sure you want to delete {{contactName}} and all {{channelsCount}} of their conversation history? To confirm, type {{confirmationText}} in the field below.",
```

And then pass `t('Delete')` as a interpolation prop to the string.